server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/luno/LunoExchange.java

```diff
@@ -0,0 +1,150 @@
+import java.util.Set;
+
+import si.mazi.rescu.RestProxyFactory;
+import si.mazi.rescu.ClientConfig;
+import si.mazi.rescu.ClientConfigUtil;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class LunoExchange implements IExchange {
+
+    private String preferredFiatCurrency = FiatCurrency.ZAR.getCode();
+    private String clientKey;
+    private String clientSecret;
+    private String typeorder;
+    private LunoExchangeAPI api;
+    private final Logger log;
+
+    public LunoExchange(String clientKey, String clientSecret, String preferredFiatCurrency, String typeorder) {
+        this.preferredFiatCurrency = FiatCurrency.ZAR.getCode();
+        this.clientKey = clientKey;
+        this.clientSecret = clientSecret;
+        this.typeorder = typeorder;
+        log = LoggerFactory.getLogger("batm.master.exchange.luno");
+        final ClientConfig config = new ClientConfig();
+        ClientConfigUtil.addBasicAuthCredentials(config, clientKey, clientSecret);
+        api = RestProxyFactory.createProxy(LunoExchangeAPI.class, "https://api.mybitx.com", config);
+    }
+
+
+    @Override
+    public Set<String> getCryptoCurrencies() {
+        Set<String> cryptoCurrencies = new HashSet<>();
+        cryptoCurrencies.add(CryptoCurrency.BTC.getCode());
+        return cryptoCurrencies;
+    }
+
+    @Override
+    public Set<String> getFiatCurrencies() {
+        Set<String> fiatCurrencies = new HashSet<>();
+        fiatCurrencies.add(FiatCurrency.ZAR.getCode());
+        return fiatCurrencies;
+    }
+
+    @Override
+    public BigDecimal getCryptoBalance(String cryptoCurrency) {
+        final LunoBalanceData balance = api.getBalance();
+        final BigDecimal cryptoballance = balance.getBalance("XBT");
+        log.debug("{} exbalance = {}", cryptoCurrency, cryptoballance);
+        return cryptoballance;
+    }
+
+    @Override
+    public String getDepositAddress(String cryptoCurrency) {
+        if (cryptoCurrency.equals("BTC")) {
+            final LunoAddressData address = api.getAddress("XBT");
+            return address.getAddress();
+        } else {
+            final LunoAddressData address = api.getAddress(cryptoCurrency);
+            return address.getAddress();
+        }
+    }
+
+    @Override
+    public String getPreferredFiatCurrency() {
+        return this.preferredFiatCurrency;
+    }
+    
+    
+    @Override
+    public BigDecimal getFiatBalance(String fiatCurrency) {
+        final LunoBalanceData balance = api.getBalance();
+        final BigDecimal fiatballance = balance.getBalance("ZAR");
+        log.debug("{} exbalance = {}", fiatCurrency, fiatballance);
+        return fiatballance;
+    }
+    
+    
+    @Override
+    public String purchaseCoins(BigDecimal amount, String cryptoCurrency, String fiatCurrencyToUse, String description) {
+        String type = "BUY";
+        String pair = "XBTZAR";
+        final LunoTickerData btcZar = api.getTicker("XBTZAR");
```

Comment:
Build currency pair based on cryptoCurrency, and fiatCurrencyToUse. Remove hardcoded XBT and ZAR.